examples/fail/fail.bal

```diff
@@ -0,0 +1,33 @@
+import ballerina/io;
+
+const ACCOUNT_NOT_FOUND = "AccountNotFound";
+
+const INVALID_ACCOUNT_ID = "InvalidAccountID";
+
+public function main() {
+    int|error negResult = getAccountBalance(-1);
+    if (negResult is error) {
+        io:println("Error returned:", negResult.message());
+    }
+
+    int|error invalidResult = getAccountBalance(200);
+    if (invalidResult is error) {
+        io:println("Error returned:", invalidResult.message());
+    }
+}
+
+function getAccountBalance(int accountID) returns int|error {
+     if (accountID < 0) {
+         // Creates an error with "InvalidAccountID" as the reason if
+         //`accountID` is less than zero.
+         error invalidAccoundIdError = error(INVALID_ACCOUNT_ID, accountID = accountID);
+         // Returns the error. Invokable return type should be assignable to the returned error type.
```

Comment:
```suggestion
         // Returns the error. An invokable return type should be assigned to the returned error type.
```

Reply:
What I intended was to say that the error returned by fail statement should be assignable to the invokable's return type

Reply:
Shall we say "The error returned by the `fail` statement should be assignable to the return type of the invoked object."?